File: c/abc039_c.py
# 6WA やべ〜
import sys


def input(): return sys.stdin.readline().strip()
def I(): return int(input())
def LI(): return list(map(int, input().split()))
def IR(n): return [I() for i in range(n)]
def LIR(n): return [LI() for i in range(n)]
def SR(n): return [S() for i in range(n)]
def S(): return input()
def LS(): return input().split()


INF = float('inf')

s = S()
base = 'WBWBWWBWBWBW' + 'WBWBWWBWBWBW' + 'WBWBWWBWBWBW'
anses = {0: 'Do', 2: 'Re', 4: 'Mi', 5: 'Fa', 7: 'So', 9: 'La', 11: 'Si'}
ans = ''
for i in [0, 2, 4, 5, 7, 9, 11]:
    if s == base[i:i + 20]:
        ans = anses[i]
        break
print(ans)

File: c/abc039_c.js
// 51
const main = (arg) => {
  const lines = arg.trim().split('\n');
  const S = lines[0];
  let base = 'WBWBWWBWBWBW' + 'WBWBWWBWBWBW' + 'WBWBWWBWBWBW';
  let scales = new Map([
    [0, 'Do'],
    [2, 'Re'],
    [4, 'Mi'],
    [5, 'Fa'],
    [7, 'So'],
    [9, 'La'],
    [11, 'Si'],
  ]);
  let ans;

  for (const i of scales.keys()) {
    // console.log(i);
    // console.log(base.slice(i, i + 20));
    if (base.slice(i, i + 20) === S) {
      ans = scales.get(i);
      break;
    }
  }
  console.log(ans);
};
main(require('fs').readFileSync('/dev/stdin', 'utf8'));
